packages/shared/src/ShadowRoot/createReactRootShadowed.tsx

```diff
@@ -31,6 +31,22 @@ export interface ReactRootShadowed {
     // do not name it as unmount otherwise it might be compatible with ReactDOM's Root interface.
     destory(): void
 }
+let observing = false
+const pendingMountPair = new Map<ShadowRoot, Function>()
+const mutationObserver = new MutationObserver(() => {
```

Comment:
Is it possible to use `mutationsList` in the first parameter, then filter with `pendingMountPair` that could be faster?

Reply:
I don't think either way will affect the performance. The key point is it observes the whole document changes.